### PR TITLE
[quick_checks] Add test to ensure buildspecs have job type

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Add quick checks test to ensure that buildspecs are nested under job type
- Ensure that those buildspecs do not have the other image type mentioned
- Ignore eks buildspecs and top level buildspecs
- Remove old buildspec templates

### Tests
- [x] Test MXNet inference as sanity test for general buildspec [b88ab2f](https://github.com/aws/deep-learning-containers/pull/2639/commits/b88ab2f03626539f47625939016fff101f5cc859)
- [x] Test PyTorch inference [16b898a](https://github.com/aws/deep-learning-containers/pull/2639/commits/16b898a562cab5567888af8125d7f2b17bc0b976)
- [x] Test Neuron PT/TF inference and PT training [f27b871](https://github.com/aws/deep-learning-containers/pull/2639/commits/f27b8711f47932b39d235c1085dfaec895570967)
- [x] Test presence of framework level buildspec (expected failure) [59e406d](https://github.com/aws/deep-learning-containers/pull/2639/commits/59e406d11ad5f968d305af395961e72102197739)
- [x] Test presence of inference images in training definition (expected failure) [47ef91e](https://github.com/aws/deep-learning-containers/pull/2639/commits/47ef91e92d13345e2d75f9d7b85be0325a45f3c6)
- [x] Test improperly formed buildspec config (expected failure) [189bb33](https://github.com/aws/deep-learning-containers/pull/2639/commits/189bb3394db480cfd5f2201e34df9112f9999e03)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/sanity/quick_checks/test_buildspecs.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    for root, dirnames, filenames in os.walk(dlc_base_dir):
+        for filename in filenames:
+            if buildspec_pattern.search(filename):
+                buildspec_path = os.path.join(dlc_base_dir, root, filename)
+
+                # There are several top level buildspecs, so if we are in the 
+                # top level dir, src dir or eks infra dir, don't count these
+                if (
+                    os.path.split(buildspec_path)[0] != dlc_base_dir
+                    and "eks_infrastructure" not in buildspec_path
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/quick_checks/test_buildspecs.py:37]: is there no better way to handle this? avoid hardcoding? we need to do handle this everytime a new dir is added

Comment (arjkesh) [reply test/dlc_tests/sanity/quick_checks/test_buildspecs.py]: Not every time a new dir is added, but every time a buildspec is added without an image type.

We can put allowed dirs in a list if that's preferred.

The other thing we can do is just do a re.match on buildspec instead, but then if a team names a framework buildspec <something>-buildspec.yml, it won't catch it.

Comment (arjkesh) [reply test/dlc_tests/sanity/quick_checks/test_buildspecs.py]: Resolved by using re.match, and also checking in src/main.py for buildspec format

File: test/dlc_tests/sanity/quick_checks/test_buildspecs.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    training_pattern = re.compile(r"training", re.IGNORECASE)
+    inference_pattern = re.compile(r"inference", re.IGNORECASE)
+
+    # Navigate through files and look for matches
+    for root, dirnames, filenames in os.walk(dlc_base_dir):
+        for filename in filenames:
+            if buildspec_pattern.match(filename):
+                buildspec_path = os.path.join(dlc_base_dir, root, filename)
+
+                # Don't look for framework buildspecs in the top level directory - these are not framework buildspecs
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/quick_checks/test_buildspecs.py:35]: this is nice

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**